src/dashboardUsers/controller/edit-userByAdmin.js

```diff
@@ -0,0 +1,56 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+const UserModel = require("../../authentication/model/user-model");
+
+exports.updateUser = () => async (req, res, next) => {
+  {
+    const hashConst = 10;
+    const { _id } = req.body;
+    const userData = req.body;
+
+    const validationId = mongoose.Types.ObjectId.isValid(_id);
+    if (!validationId) {
+      return res.status(400).json({
+        error: "Sorry, but provided id is inccorect!",
+        data: [],
+      });
+    } else {
+      const foundUser = await UserModel.findOne({ _id });
+      if (!foundUser) {
+        return res.status(400).json({
+          error: "Sorry, this user doesn't exist in the database!",
+          data: [],
+        });
+      } else {
+// Here we have a conditional block that checks if the edited mail exists in the database and if the password has been changed it will be encrypted and sent like this.
+        
+        if (userData.password || userData.email) {
```

Comment:
@Murukaen This is the change I told you about. Specifically, it is a conditional block "if else" that checks if the edited email does not already exist in the database. And encrypts the password if it was edited.